releasenotes-builder/src/main/java/com/github/checkstyle/MainProcess.java

```diff
@@ -79,7 +80,50 @@ public static List<String> runPostGenerationAndPublication(
             Multimap<String, ReleaseNotesMessage> releaseNotes, CliOptions cliOptions)
             throws IOException, TemplateException {
         runPostGeneration(releaseNotes, cliOptions);
-        return runPostPublication(cliOptions);
+        final List<String> errors = new ArrayList<>();
+        errors.addAll(validateNotes(releaseNotes, cliOptions));
+        errors.addAll(runPostPublication(cliOptions));
+        return errors;
+    }
+
+    /**
+     * Validate that pom version matches issues labels in release notes.
+     *
+     * @param releaseNotes map of release notes messages.
+     * @param cliOptions command line options.
+     * @return list of notes validation errors.
+     */
+    private static List<String> validateNotes(Multimap<String,
+                                                    ReleaseNotesMessage> releaseNotes,
+                                                    CliOptions cliOptions) {
+        final String releaseVersion = cliOptions.getReleaseNumber();
+        final long amountOfCommas = getAmountOfCommasInReleaseVersion(releaseVersion);
+
+        final boolean containsNewOrBreakingCompatabilityLabel =
+            releaseNotes.containsKey(Constants.NEW_FEATURE_LABEL)
+            || releaseNotes.containsKey(Constants.NEW_MODULE_LABEL)
+            || releaseNotes.containsKey(Constants.BREAKING_COMPATIBILITY_LABEL);
+
+        final List<String> errors = new ArrayList<>();
+
+        if (amountOfCommas == 1 && containsNewOrBreakingCompatabilityLabel
+            || amountOfCommas == 2 && !containsNewOrBreakingCompatabilityLabel) {
+            errors.add("Validation of release notes failed. "
+                + "There is a mismatch between release version and issues labels.");
```

Comment:
Let's be more detailed on what is problem
To make message special for each invalid scenario.
To let maintainer not think and simply follow suggestion.

Reply:
Here are my suggestions for both. Let me know what you think:
- `isPatch(amountOfCommas) && containsNewOrBreakingCompatabilityLabel`
  - `Validation of release number failed. Release number is a patch(10.4), but release notes do not contain 'new' or 'breaking compatability' labels.`
- `isMinor(amountOfCommas) && !containsNewOrBreakingCompatabilityLabel`
  - `Validation of release number failed. Release number is minor(10.3.5), but release notes contain 'new' or 'breaking compatability' labels.`

Reply:
Yes.

`Please correct release number by running https://github.com/checkstyle/checkstyle/actions/workflows/bump-version-and-update-milestone.yml`

Reply:
FYI https://github.com/checkstyle/checkstyle/wiki/Semi-automated-release

Reply:
Done.